lms/djangoapps/teams/static/teams/js/spec/views/teams_tab_spec.js

```diff
@@ -78,6 +109,74 @@ define([
             });
         });
 
+        describe('Analytics Events', function () {
+            beforeEach(function () {
+            //    TODO: createTeamsTabView, navigate to browse teams page...
+            });
+            it('fires a page view event for the topic page', function () {
+                var requests = AjaxHelpers.requests(this),
+                    teamsTabView = createTeamsTabView();
+                teamsTabView.router.navigate('topics/' + TeamSpecHelpers.testTopicID, {trigger: true});
+                AjaxHelpers.respondWithJson(requests, {});
+                expect(Logger.log).toHaveBeenCalledWith('edx.team.page_viewed', {
+                    page_name: 'single-topic',
+                    topic_id: TeamSpecHelpers.testTopicID,
+                    team_id: null
+                });
+            });
+
+            it('fires a page view event for the team page', function () {
+                var requests = AjaxHelpers.requests(this),
+                    teamsTabView = createTeamsTabView(),
+                    team_id = 'test_team_id';
+                teamsTabView.router.navigate('teams/' + TeamSpecHelpers.testTopicID + '/' + team_id, {trigger: true});
+                AjaxHelpers.respondWithJson(requests, {});
+                expect(Logger.log).toHaveBeenCalledWith('edx.team.page_viewed', {
+                    page_name: 'single-team',
+                    topic_id: TeamSpecHelpers.testTopicID,
+                    team_id: team_id
+                });
+            });
+
+            it('fires a page view event for the search team page', function () {
+                var requests = AjaxHelpers.requests(this),
+                    teamsTabView = createTeamsTabView();
+                teamsTabView.browseTopic(TeamSpecHelpers.testTopicID);
+                teamsTabView.router.navigate('topics/' + TeamSpecHelpers.testTopicID + '/search', {trigger: true});
+                AjaxHelpers.respondWithJson(requests, {});
+                expect(Logger.log).toHaveBeenCalledWith('edx.team.page_viewed', {
+                    page_name: 'search-teams',
+                    topic_id: TeamSpecHelpers.testTopicID,
+                    team_id: null
+                });
+            });
+
+            it('fires a page view event for the new team page', function () {
+                var teamsTabView = createTeamsTabView();
+                teamsTabView.router.navigate('topics/' + TeamSpecHelpers.testTopicID + '/create-team', {trigger: true});
+                expect(Logger.log).toHaveBeenCalledWith('edx.team.page_viewed', {
+                    page_name: 'new-team',
+                    topic_id: TeamSpecHelpers.testTopicID,
+                    team_id: null
+                });
+            });
+
+            it('fires a page view event for the edit team page', function () {
```

Comment:
Would it be reasonable to simplify all these tests by doing something DDT-ish? You could loop over a list of `(route, event)` pairs, navigate to the `route`, and then assert that `event` has been emitted. I think the de-duplication of code is worth it.